src/NearestNeighborModels.jl

```diff
@@ -57,9 +57,18 @@ const KNNClassifierDescription = """
 const KNNCoreFields = """
     * `K::Int=5` : number of neighbors
     * `algorithm::Symbol = :kdtree` : one of `(:kdtree, :brutetree, :balltree)`
-    * `metric::Metric = Euclidean()` : a `Metric` object for the distance between points
-    * `leafsize::Int = 10` : at what number of points to stop splitting the tree
-    * `reorder::Bool = true` : if true puts points close in distance close in memory
+    * `metric::Metric = Euclidean()` : any `Metric` from 
+        [Distances.jl](https://github.com/JuliaStats/Distances.jl) for the 
+        distance between points. For `algorithm = :kdtree` only metrics which are of 
+        type `$(NN.MinkowskiMetric)` are supported.
+    * `leafsize::Int = algorithm == :brutetree ? 0 : 10` : determines the number of points 
```

Comment:
This way of writing the default is a little cryptic for the general user.  Maybe just write `leafsize::Int == 10` and just say this is ignored if  `algorithm=:brutetree`.